File: blaze_flutter_sdk/android/src/main/kotlin/com/blaze/flutterblazesdk/players/moments/BlazeReactMomentsPlayerStyle.kt
package com.blaze.flutterblazesdk.players.moments

import androidx.annotation.Keep
import com.blaze.blazesdk.style.players.BlazePlayerDisplayMode
import com.blaze.blazesdk.style.players.moments.BlazeMomentsPlayerBodyTextStyle
import com.blaze.blazesdk.style.players.moments.BlazeMomentsPlayerCtaIconStyle
import com.blaze.blazesdk.style.players.moments.BlazeMomentsPlayerCtaStyle
import com.blaze.blazesdk.style.players.moments.BlazeMomentsPlayerFooterGradientStyle
import com.blaze.blazesdk.style.players.moments.BlazeMomentsPlayerHeadingTextStyle
import com.blaze.blazesdk.style.players.moments.BlazeMomentsPlayerStyle
import com.blaze.flutterblazesdk.players.shared.BlazeReactFirstTimeSlideCTAStyle
import com.blaze.flutterblazesdk.players.shared.BlazeReactFirstTimeSlideInstructionStyle
import com.blaze.flutterblazesdk.players.shared.BlazeReactFirstTimeSlideTextStyle
import com.blaze.flutterblazesdk.players.shared.BlazeReactPlayerButtonStyle
import com.blaze.flutterblazesdk.players.shared.BlazeReactSeekBarStyle
import com.blaze.flutterblazesdk.utils.parsing.BlazeReactColor
import com.blaze.flutterblazesdk.utils.parsing.BlazeReactImage
import com.blaze.flutterblazesdk.utils.parsing.BlazeReactTitleFont
import com.blaze.flutterblazesdk.utils.parsing.gson.BlazeEnumMapper
import com.blaze.flutterblazesdk.widgets.customization.BlazeReactMargins

@Keep
data class BlazeReactMomentsPlayerStyle(
    val headingText: BlazeReactMomentsPlayerHeadingTextStyle?,
    val bodyText: BlazeReactMomentsPlayerBodyTextStyle?,
    val buttons: BlazeReactMomentsPlayerButtonsStyle?,
    val chips: BlazeReactMomentsPlayerChipsStyle?,
    val backgroundColor: String?, // Hex
    val cta: BlazeReactMomentsPlayerCtaStyle?,
    val headerGradient: BlazeReactMomentsPlayerHeaderGradientStyle?,
    val footerGradient: BlazeReactMomentsPlayerFooterGradientStyle?,
    val firstTimeSlide: BlazeReactMomentsPlayerFirstTimeSlideStyle?,
    val seekBar: BlazeReactMomentsPlayerSeekBarStyle?,
    val bottomComponentsAlignment: BlazeReactBottomComponentsAlignment?,
    var playerDisplayMode: BlazeReactPlayerDisplayMode?
)

@Keep
data class BlazeReactMomentsPlayerHeadingTextStyle(
    val textSize: Float?,
    val textColor: String?, // Hex
    val font: BlazeReactTitleFont?,
    val contentSource: ContentSource?,
    val isVisible: Boolean?
) {
    @Keep
    enum class ContentSource(override val rawValue: String) :
        BlazeEnumMapper<BlazeMomentsPlayerHeadingTextStyle.BlazeContentSource> {
        TITLE("Title") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerHeadingTextStyle.BlazeContentSource.TITLE
        },
        SUBTITLE("Subtitle") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerHeadingTextStyle.BlazeContentSource.SUBTITLE
        }
    }
}

@Keep
data class BlazeReactMomentsPlayerBodyTextStyle(
    val textSize: Float?,
    val textColor: String?, // Hex
    val font: BlazeReactTitleFont?,
    val contentSource: ContentSource?,
    val isVisible: Boolean?
) {
    @Keep
    enum class ContentSource(override val rawValue: String) :
        BlazeEnumMapper<BlazeMomentsPlayerBodyTextStyle.BlazeContentSource> {
        TITLE("Title") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerBodyTextStyle.BlazeContentSource.TITLE
        },
        SUBTITLE("Subtitle") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerBodyTextStyle.BlazeContentSource.SUBTITLE
        },
        DESCRIPTION("Description") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerBodyTextStyle.BlazeContentSource.DESCRIPTION
        }
    }
}

@Keep
data class BlazeReactMomentsPlayerButtonsStyle(
    val mute: BlazeReactPlayerButtonStyle?,
    val exit: BlazeReactPlayerButtonStyle?,
    val share: BlazeReactPlayerButtonStyle?,
    val like: BlazeReactPlayerButtonStyle?,
    val play: BlazeReactPlayerButtonStyle?
)

@Keep data class BlazeReactMomentsPlayerChipsStyle(val ad: BlazeReactMomentsPlayerChipStyle?)

@Keep
data class BlazeReactMomentsPlayerChipStyle(
    val titlePadding: BlazeReactMargins?,
    val text: String?,
    val textColor: String?,
    val backgroundColor: String?,
    var isVisible: Boolean?
)

@Keep
data class BlazeReactMomentsPlayerCtaStyle(
    val cornerRadius: Int?,
    val textSize: Float?,
    val font: BlazeReactTitleFont?,
    val width: Int?,
    val height: Int?,
    val icon: BlazeReactMomentsPlayerCtaIconStyle?,
    val layoutPositioning: CTAPositioning?,
    val horizontalAlignment: CTAHorizontalAlignment?
) {
    @Keep
    enum class CTAPositioning(override val rawValue: String) :
        BlazeEnumMapper<BlazeMomentsPlayerCtaStyle.BlazeCTAPositioning> {
        CTA_BELLOW_BOTTOM_BUTTONS_BOX("CtaBellowBottomButtonsBox") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerCtaStyle.BlazeCTAPositioning.CTA_BELLOW_BOTTOM_BUTTONS_BOX
        },
        CTA_NEXT_TO_BOTTOM_BUTTONS_BOX("CtaNextToBottomButtonsBox") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerCtaStyle.BlazeCTAPositioning.CTA_NEXT_TO_BOTTOM_BUTTONS_BOX
        }
    }

    @Keep
    enum class CTAHorizontalAlignment(override val rawValue: String) :
        BlazeEnumMapper<BlazeMomentsPlayerCtaStyle.BlazeCTAHorizontalAlignment> {
        START("Start") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerCtaStyle.BlazeCTAHorizontalAlignment.START
        },
        CENTER("Center") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerCtaStyle.BlazeCTAHorizontalAlignment.CENTER
        },
        END("End") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerCtaStyle.BlazeCTAHorizontalAlignment.END
        },
        FULL_AVAILABLE_WIDTH("FullAvailableWidth") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerCtaStyle.BlazeCTAHorizontalAlignment.FULL_AVAILABLE_WIDTH
        }
    }
}

@Keep
data class BlazeReactMomentsPlayerHeaderGradientStyle(
    val isVisible: Boolean?,
    val startColor: String?, // Hex
    val endColor: String?, // Hex
)

@Keep
data class BlazeReactMomentsPlayerFooterGradientStyle(
    val isVisible: Boolean?,
    val startColor: String?, // Hex
    val endColor: String?, // Hex
    val endPositioning: EndPositioning?
) {
    @Keep
    enum class EndPositioning(override val rawValue: String) :
        BlazeEnumMapper<BlazeMomentsPlayerFooterGradientStyle.BlazeEndPositioning> {
        BOTTOM_TO_PLAYER("BottomToPlayer") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerFooterGradientStyle.BlazeEndPositioning.BOTTOM_TO_PLAYER
        },
        BOTTOM_TO_CONTAINER("BottomToContainer") {
            override fun mapToBlazeEnumClass() =
                BlazeMomentsPlayerFooterGradientStyle.BlazeEndPositioning.BOTTOM_TO_CONTAINER
        }
    }
}

@Keep
data class BlazeReactMomentsPlayerFirstTimeSlideStyle(
    val show: Boolean?,
    val cta: BlazeReactFirstTimeSlideCTAStyle?,
    val backgroundColor: BlazeReactColor?,
    val mainTitle: BlazeReactFirstTimeSlideTextStyle?,
    val subtitle: BlazeReactFirstTimeSlideTextStyle?,
    val instructions: BlazeReactMomentsPlayerFirstTimeSlideInstructionsStyle?
)

@Keep
data class BlazeReactMomentsPlayerFirstTimeSlideInstructionsStyle(
    val next: BlazeReactFirstTimeSlideInstructionStyle?,
    val pause: BlazeReactFirstTimeSlideInstructionStyle?,
    val previous: BlazeReactFirstTimeSlideInstructionStyle?,
    val play: BlazeReactFirstTimeSlideInstructionStyle?
)

@Keep
data class BlazeReactMomentsPlayerSeekBarStyle(
    val isVisible: Boolean?,
    val playingState: BlazeReactSeekBarStyle?,
    val pausedState: BlazeReactSeekBarStyle?,
    val horizontalSpacing: Int?,
    val bottomSpacing: Int?
)

@Keep
data class BlazeReactMomentsPlayerCtaIconStyle(
    val iconImage: BlazeReactImage?,
    val iconPositioning: BlazeReactMomentsCTAIconPositioning?,
    val iconTint: String?
)

@Keep
enum class BlazeReactBottomComponentsAlignment(override val rawValue: String) :
    BlazeEnumMapper<BlazeMomentsPlayerStyle.BlazeBottomComponentsAlignment> {
    RELATIVE_TO_CONTAINER("RelativeToContainer") {
        override fun mapToBlazeEnumClass() =
            BlazeMomentsPlayerStyle.BlazeBottomComponentsAlignment.RELATIVE_TO_CONTAINER
    },
    RELATIVE_TO_PLAYER("RelativeToPlayer") {
        override fun mapToBlazeEnumClass() =
            BlazeMomentsPlayerStyle.BlazeBottomComponentsAlignment.RELATIVE_TO_PLAYER
    },
    FIT_CTA_BELOW_PLAYER("FitCtaBelowPlayer") {
        override fun mapToBlazeEnumClass() =
            BlazeMomentsPlayerStyle.BlazeBottomComponentsAlignment.FIT_CTA_BELOW_PLAYER
    }
}

@Keep
enum class BlazeReactPlayerDisplayMode(override val rawValue: String) :
    BlazeEnumMapper<BlazePlayerDisplayMode> {
    FIXED_RATIO_9_16("FixedRatio_9_16") {
        override fun mapToBlazeEnumClass() = BlazePlayerDisplayMode.FIXED_RATIO_9_16
    },
    RESIZE_ASPECT_FILL_CENTER_CROP("ResizeAspectFillCenterCrop") {
        override fun mapToBlazeEnumClass() = BlazePlayerDisplayMode.RESIZE_ASPECT_FILL_CENTER_CROP
    }
}

@Keep
enum class BlazeReactMomentsCTAIconPositioning(override val rawValue: String) :
    BlazeEnumMapper<BlazeMomentsPlayerCtaIconStyle.BlazeIconPositioning> {
    START("Start") {
        override fun mapToBlazeEnumClass() =
            BlazeMomentsPlayerCtaIconStyle.BlazeIconPositioning.START
    }
}

File: blaze_flutter_sdk/ios/Sources/Players/Moments/BlazeReactMomentsPlayerStyle.swift
//
//  BlazeReactMomentsPlayerStyle.swift
//  blaze_flutter_sdk
//
//  Copied exactly from React Native BlazeReactMomentsPlayerStyle.swift
//

import Foundation

struct BlazeReactMomentsPlayerStyle: Codable {
    var headingText: BlazeReactMomentsPlayerHeadingTextStyle?
    var bodyText: BlazeReactMomentsPlayerBodyTextStyle?
    var buttons: BlazeReactMomentsPlayerButtonsStyle?
    var chips: BlazeReactMomentsPlayerChipsStyle?
    var backgroundColor: String?
    var cta: BlazeReactMomentsPlayerCtaStyle?
    var headerGradient: BlazeReactMomentsPlayerHeaderGradientStyle?
    var footerGradient: BlazeReactMomentsPlayerFooterGradientStyle?
    var firstTimeSlide: BlazeReactMomentsPlayerFirstTimeSlideStyle?
    let seekBar: BlazeReactMomentsPlayerSeekBarStyle?
    var bottomComponentsAlignment: BlazeReactBottomComponentsAlignment?
    var playerDisplayMode: BlazeReactPlayerDisplayMode?
}

struct BlazeReactMomentsPlayerHeadingTextStyle: Codable {
    var textSize: CGFloat? // Only Android
    var font: BlazeReactTitleFont?
    var textColor: String?
    var contentSource: ContentSource?
    var isVisible: Bool?
    
    enum ContentSource: String, Codable {
        case title = "Title"
        case subtitle = "Subtitle"
    }
}

struct BlazeReactMomentsPlayerBodyTextStyle: Codable {
    var textSize: CGFloat?
    var font: BlazeReactTitleFont?
    var textColor: String?
    var contentSource: ContentSource?
    var isVisible: Bool?
    
    enum ContentSource: String, Codable {
        case title = "Title"
        case subtitle = "Subtitle"
        case description = "Description"
    }
}

struct BlazeReactMomentsPlayerButtonsStyle: Codable {
    var mute: BlazeReactPlayerButtonStyle?
    var exit: BlazeReactPlayerButtonStyle?
    var share: BlazeReactPlayerButtonStyle?
    var like: BlazeReactPlayerButtonStyle?
    var play: BlazeReactPlayerButtonStyle?
}

struct BlazeReactMomentsPlayerChipsStyle: Codable {
    var ad: BlazeReactMomentsPlayerChipStyle?
}

struct BlazeReactMomentsPlayerChipStyle: Codable {
    var titlePadding: BlazeReactMargins?
    var text: String?
    var textColor: String?
    var backgroundColor: String?
    var isVisible: Bool?
}

struct BlazeReactMomentsPlayerCtaStyle: Codable {
    var cornerRadius: CGFloat?
    var width: CGFloat?
    var height: CGFloat?
    var font: BlazeReactTitleFont?
    var textSize: CGFloat? // Only for Android
    var layoutPositioning: CTAPositioning?
    var icon: BlazeReactMomentCtaIcon?
    var horizontalAlignment: CTAHorizontalAlignment?
    
    enum CTAPositioning: String, Codable {
        case bellowBottomButtonsBox = "CtaBellowBottomButtonsBox"
        case nextToBottomButtonsBox = "CtaNextToBottomButtonsBox"
    }
    
    enum CTAHorizontalAlignment: String, Codable {
        case start = "Start"
        case center = "Center"
        case end = "End"
        case fullAvailableWidth = "FullAvailableWidth"
    }
}

struct BlazeReactMomentsPlayerHeaderGradientStyle : Codable {
    var isVisible: Bool?
    var startColor: String?
    var endColor: String?
}

struct BlazeReactMomentsPlayerFooterGradientStyle : Codable {
    var isVisible: Bool?
    var startColor: String?
    var endColor: String?
    var endPositioning: EndPositioning?
    
    enum EndPositioning: String, Codable {
        case bottomToPlayer = "BottomToPlayer"
        case bottomToContainer = "BottomToContainer"
    }
}

struct BlazeReactMomentsPlayerFirstTimeSlideStyle: Codable {
    var show: Bool?
    var cta: BlazeReactFirstTimeSlideCTAStyle?
    var backgroundColor: BlazeReactColor?
    var mainTitle: BlazeReactFirstTimeSlideTextStyle?
    var subtitle: BlazeReactFirstTimeSlideTextStyle?
    var instructions: BlazeReactMomentsPlayerFirstTimeSlideInstructionsStyle?
}

struct BlazeReactMomentsPlayerFirstTimeSlideInstructionsStyle: Codable{
    var next: BlazeReactFirstTimeSlideInstructionStyle?
    var previous: BlazeReactFirstTimeSlideInstructionStyle?
    var pause: BlazeReactFirstTimeSlideInstructionStyle?
    var play: BlazeReactFirstTimeSlideInstructionStyle?
}

struct BlazeReactMomentsPlayerSeekBarStyle: Codable {
    let isVisible: Bool?
    let playingState: BlazeReactSeekBarStyle?
    let pausedState: BlazeReactSeekBarStyle?
    let horizontalSpacing: CGFloat?
    let bottomSpacing: CGFloat?
}

struct BlazeReactMomentCtaIcon: Codable {
    var iconImage: BlazeReactImage?
} 